### PR TITLE
Unit-test that `class NAME VERSION :ATTRS` syntax works

### LINKED ISSUE #20891: Class declaration with both version and :isa fails to compile
**Description**

A class declaration that has both a declared version and an `:isa` attribute isn’t parsed correctly. Compilation fails complaining about a non-numeric version format. The class block syntax and the unit class syntax are both affected.

```perl
class A 1.23         { }  # works
class B 1.23 :isa(A) { }  # fails: Invalid version format (non-numeric data)

class A 1.23         ;    # works
class B 1.23 :isa(A) ;    # fails: Invalid version format (non-numeric data)
```

While [perlclass](https://github.com/Perl/perl5/blob/b45c1f8aeb97b765fe1c30fe69a0b0bbff9d9925/pod/perlclass.pod) doesn’t actually say _explicitly_ that this should work, it does seem to me that it was intended to do so.

I found the following comment by @leonerd for the `package` syntax and am wondering if it might be relevant here:

https://github.com/Perl/perl5/blob/ae790582414fc0cbc81a9f1a71fc36e055d34d86/perly.y#L435-L439

**Steps to Reproduce**
```sh
perl -Mfeature=class -e 'class A 1.23 {} class B 1.23 :isa(A) {}'
```

**Expected behaviour**

Compilation succeeds.

**Perl configuration**
```
Summary of my perl5 (revision 5 version 37 subversion 9) configuration:
   
  Platform:
    osname=darwin
    osvers=19.6.0
    archname=darwin-ld-2level
    uname='darwin solway.local 19.6.0 darwin kernel version 19.6.0: tue jun 21 21:18:39 pdt 2022; root:xnu-6153.141.66~1release_x86_64 x86_64 '
    config_args='-de -Duselongdouble -Dprefix=/Users/aj/.perlbrew/perls/perl-5.37.9 -Dusedevel -Aeval:scriptdir=/Users/aj/.perlbrew/perls/perl-5.37.9/bin'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=define
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.15 -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    optimize='-O3'
    cppflags='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.15 -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='Apple LLVM 12.0.0 (clang-1200.0.32.27)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='long double'
    nvsize=16
    Off_t='off_t'
    lseeksize=8
    alignbytes=16
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -mmacosx-version-min=10.15 -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/lib /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib /Library/Developer/CommandLineTools/usr/lib /usr/lib
    libs=-lpthread -lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-lpthread -ldl -lm -lutil -lc
    libc=
    so=dylib
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=bundle
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags=' '
    lddlflags=' -mmacosx-version-min=10.15 -bundle -undefined dynamic_lookup -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_HASH_FUNC_SIPHASH13
    PERL_HASH_USE_SBOX32
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_USE_DEVEL
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_LONG_DOUBLE
    USE_PERLIO
    USE_PERL_ATOF
  Built under darwin
  Compiled at Feb 21 2023 03:20:59
  %ENV:
    PERLBREW_HOME="/Users/aj/.perlbrew"
    PERLBREW_MANPATH="/Users/aj/.perlbrew/perls/perl-5.37.9/man"
    PERLBREW_PATH="/Users/aj/.perlbrew/bin:/Users/aj/.perlbrew/perls/perl-5.37.9/bin"
    PERLBREW_PERL="perl-5.37.9"
    PERLBREW_ROOT="/Users/aj/.perlbrew"
    PERLBREW_SHELLRC_VERSION="0.92"
    PERLBREW_VERSION="0.92"
    PERL_CPANM_OPT="--from https://www.cpan.org/"
  @INC:
    /Users/aj/.perlbrew/perls/perl-5.37.9/lib/site_perl/5.37.9/darwin-ld-2level
    /Users/aj/.perlbrew/perls/perl-5.37.9/lib/site_perl/5.37.9
    /Users/aj/.perlbrew/perls/perl-5.37.9/lib/5.37.9/darwin-ld-2level
    /Users/aj/.perlbrew/perls/perl-5.37.9/lib/5.37.9
```

### PR DESCRIPTION
**Pull request by leonerd**
Fixes #20891

### CODE REVIEW

**Review by rwp0 — APPROVED** [review]

**Comment by Leont** [comment]
~~~
/* This file is part of the "version" CPAN distribution.  Please avoid
   editing it in the perl core. */
~~~

**Comment by leonerd** [comment]
> ```
> /* This file is part of the "version" CPAN distribution.  Please avoid
>    editing it in the perl core. */
> ```

Oh. Huh. It was already edited as part of the original feature-class branch anyway. Should that get backported in?

**Comment by tonycoz** [comment]
It says "avoid editing" rather than "don't edit", and I think this change is reasonable.

We've edited it in the past to support core changes.

**Review by tonycoz — APPROVED** [review]

**Comment by Leont** [comment]
> It says "avoid editing" rather than "don't edit", and I think this change is reasonable.
>
> We've edited it in the past to support core changes.

It's not changing in core that I mind, it's the fact that it will probably be me who has to clean things up later.

**Comment by leonerd** [comment]
Raised on https://github.com/Perl/version.pm/pull/35

**Comment by rwp0** [comment]
> Raised on https://github.com/Perl/version.pm/pull/35

Can this PR be closed then?

**Comment by Leont** [comment]
> Can this PR be closed then?

No, it needs to be rebased after version has been synced from cpan

**Comment by leonerd** [comment]
The update to `version.pm` fixed the bug, but this PR still adds a unit test to demonstrate that the bug is now fixed.

**Review by tonycoz — APPROVED** [review]

File: t/class/inherit.t [hunk 1]
```diff
@@ -105,4 +105,12 @@ use A::B;
 EOS
 }
 
+{
+    # https://github.com/Perl/perl5/issues/20891
+    class Test6A 1.23 {}
+    class Test6B 1.23 :isa(Test6A) {}
+
+    pass('class declaration permits VERSION and :ATTRS...');
```
Comment (ilmari) [inline t/class/inherit.t]: Maybe test `ok(Test6B->new isa Test6A)` and `is(Test6B->VERSION, 1.23)` as well, to make sure the everything actually works, not just parses?

Comment (leonerd) [reply t/class/inherit.t]: Ah yes. Test added.

**Review by ilmari — APPROVED** [review]

**Pull request merged**